### DISCUSSION TITLE
JK BMS B2A20S20P

### DISCUSSION THREAD

**Discussion by maxx8888** [Louisvdw/dbus-serialbattery#535] [General]
Hi,

It seems like Charging allowed, Discharging allowed is not reported correctly to Venus OS.

Discharging disabled on Mobile App via Bluetooth -> Venos OS reports Charging allowed: No
Charging disabled on Mobile App via Bluetooth -> Venos OS doesn't report any Change.

Does somebody else face this Issue?

Venus OS: v2.94 (20230412212730)
dbus-serialbattery: 0.14.3
BMS Type: B2A20S20P
BMS Hardware Version: V11.XW
BMS Software Version: V11.25
Version: V4.10.1

**Comment by mr-manuel** [comment]
What is your Venus OS, dbus-serialbattery and BMS version?

**Reply by maxx8888** [reply]
Updated my description above.

**Reply by mr-manuel** [reply]
This should be fixed with the latest version. Currently it's already in the `jkbms_ble` fork. If you want to test it, rename your current `dbus-serialbattery` folder as backup and then run
```bash
wget -O /tmp/install-nightly.sh https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/jkbms_ble/etc/dbus-serialbattery/install-nightly.sh && bash /tmp/install-nightly.sh
```
and select `2` to install the latest script. Pay attention that the config is now done in the `config.ini` in order to persist updates.

**Reply by maxx8888** [reply]
Uh, great, thanks a lot, i will give it a try.

**Reply by mr-manuel** [reply]
Any updates on this?

**Reply by maxx8888** [reply]
didn‘t find time yet. Will try to finish the next days…

**Reply by maxx8888** [reply]
Upgraded to nightly build today.
"Allow to balance" -> works
"Allow discharge" -> works
"Allow charge" -> always true
